### DISCUSSION TITLE
[Closed] Reorganize our repo structure to be more MAUI

### DISCUSSION THREAD

**Discussion by pictos** [CommunityToolkit/Maui#208] [New Feature Discussions]
Since this repo will implement features in the same way that MAUI does, we should follow the .NET MAUI structure. We should split our code into more projects:

- MCT.Core: Where interface and common code will live
- MCT.Controls: Where the implementation of the controls will live

This separation will help us to respect the separation concerns and support more MAUI-based frameworks in the future with minimal efforts, like Comet or Fabulous.

**Comment by VladislavAntonyuk** [comment]
How do we plan to publish them? Will it be multiple packages and users install only what they need (maybe we can add `<UseCommunityToolkit>` to use all of them) Or will it be single nuget package with multiple libs?

**Reply by pictos** [reply]
Right now I think that `single nuget package with multiple libs` will do the job

**Comment by VladislavAntonyuk** [comment]
Looks like we have a similar discussion https://github.com/CommunityToolkit/Maui/discussions/175 and it received 3 dislikes

**Comment by TheCodeTraveler** [comment]
> This separation will help us to respect the separation concerns and support more MAUI-based frameworks in the future with minimal efforts, like Comet or Fabulous 

Looking at the [Comet source code](https://github.com/dotnet/comet), I'm not sure how much code we would be able to share between `CommunityToolkit.Maui` and `CommunityToolkit.Comet`. 

For example, Comet doesn't use Converters (that's mostly an MVVM/XAML thing), Comet doesn't have Behaviors, Comet uses its own unique Animation implementation and Comet creates its own Controls (e.g. it provides its own implements of `IButton`, `IImage`, etc.).

**Reply by pictos** [reply]
@brminnick for comet we can share the Views, if they are built on top of interfaces comet can generate the control using Source Generators. I know that Fabulous also use Source Generators. I'll bring @clancey to this discussion since he can give us more inputs.

**Reply by pictos** [reply]
And even if we can't share code with other MAUI frameworks that separation would be good for the project architecture, making sure that we don't mess with dependencies. In other words, `csproj`s can't have cycle references so we will make sure that `MCT.Controls` has a dependecy on `MCT.Core` and `MCT.Core` **doesn't** has a reference on `MCT.Controls`.

**Comment by CliffAgius** [comment]
Closed as answered.
